migration-guides/0.14-Migration-Guide.md

```diff
@@ -0,0 +1,597 @@
+* Unsafe global functions (i.e. those which lived in `Global.Unsafe`) have been removed altogether.
+
+**To fix:**
+
+You should remove the `globals` dependency from your library or application.
+
+If you used any `Number`-related code from the `globals` package, then you should add a new dependency on `numbers` and rename your imports:
+
+```diff
+- import Global (isNan, nan, isFinite, infinity)
++ import Data.Number (isNan, nan, isFinite, infinity)
+ 
+- import Global (toFixed, toPrecision, toExponential)
++ import Data.Number.Format (toStringWith, fixed, precision, exponential)
+
+  -- toFixed, toPrecision, and toExponential can be replaced as follows:
+- x = toFixed someInt 4.0 
++ x = toStringWith (fixed someInt) 4.0
+ 
+- y = toPrecision someInt 4.0
++ y = toStringWith (precision someInt) 4.0
+ 
+- z = toExponential intArg 4.0
++ z = toStringWith (exponential someInt) 4.0
+```
+
+If you used `readInt`, then you should add a dependency on `integers` and either use `Data.Int.fromString` if you were parsing a base-10 integer, or use `Data.Int.fromStringAs` given a `Data.Int.Radix` if you were parsing a number with another base:
+
+```diff
+- import Global (readInt)
++ import Data.Int (binary, fromStringAs)
+
+  fromBinaryString :: String -> Maybe Int
+- fromBinaryString = readInt 2
++ fromBinaryString = fromStringAs binary
+```
+
+If you used URI encoding and decoding functions, then you should add a dependency on `js-uri` and update your imports:
+
+```diff
+- import Global (decodeURI, decodeURIComponent, encodeURI, encodeURIComponent)
++ import JSURI (decodeURI, decodeURIComponent, encodeURI, encodeURIComponent)
+```
+
+If you used various unsafe functions exposed by the `globals` package, you can in most cases use functions from the `partials` package (like `unsafePartial`) combined with partial functions like `Data.Maybe.fromJust` to produce an unsafe function from a safe one.
+
+#### The dependency graph for `functors` and related packages have changed in order to unify newtypes among the libraries.
+
+Implemented in [purescript-foldable-traversable#131](https://github.com/purescript/purescript-foldable-traversable/pull/131), [purescript-functors#31](https://github.com/purescript/purescript-functors/pull/31), [purescript-profunctor#41](https://github.com/purescript/purescript-profunctor/pull/41), [purescript-distributive#17](https://github.com/purescript/purescript-distributive/pull/17), [purescript-contravariant#30](https://github.com/purescript/purescript-contravariant/pull/30), [purescript-bifunctors#22](https://github.com/purescript/purescript-bifunctors/pull/22), [purescript-tuples#46](https://github.com/purescript/purescript-tuples/pull/46), [purecsript-identity#26](https://github.com/purescript/purescript-identity/pull/26), [purescript-either#64](https://github.com/purescript/purescript-either/pull/64), and [purecsript-const#19](https://github.com/purescript/purescript-const/pull/19). 
```

Comment:
I think it might be better not to link to these PRs, especially as we aren't doing it elsewhere in this document. I'm not really sure how much these add and we don't want people thinking they need to read through these before they attempt to migrate to 0.14.

Reply:
Never mind